### PR TITLE
Chore - update tsconfig target ES version to ES2022 aligning with NodeJS 18+

### PR DESCRIPTION
**Pull request by devinea**
Update ES target to ES2022.
- output Javascript is smaller and easier to debug and read. 
- No longer need to support older runtime envs.

### CODE REVIEW

**Review by IainSAP — APPROVED** [review]
Great!

**Review by Klaus-Keller — APPROVED** [review]
Thanks @devinea, switch to `ES2022` looks good, approved from my side.

File: packages/axios-extension/test/abap/abap-service-provider.test.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ describe('AbapServiceProvider', () => {
             const catalog = provider.catalog(ODataVersion.v2);
             expect(catalog).toBeDefined();
             expect(catalog.defaults.baseURL).toBe(`${server}${V2CatalogService.PATH}`);
-            expect(provider.catalog(ODataVersion.v2)).toEqual(catalog);
+            expect(provider.catalog(ODataVersion.v2).services).toEqual(catalog.services);
```
Comment (tobiasqueck) [inline packages/axios-extension/test/abap/abap-service-provider.test.ts:147]: curious: why

Comment (devinea) [reply packages/axios-extension/test/abap/abap-service-provider.test.ts:147]: With the original code we get a failure as below. 
I assume we are really only interested in the services returned from the catalog call.
Screenshot 2024-06-24 at 07 05 05

**Review by IainSAP — APPROVED** [review]
Thx @devinea 

- es version update only, changes as expected

File: packages/cds-odata-annotation-converter/tsconfig.json [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
-    "rootDir": "src",
-    "lib": ["ES2021"]
-  },
+    "extends": "../../tsconfig.json",
+    "compilerOptions": {
+        "incremental": true,
+        "outDir": "dist",
+        "rootDir": "src",
+        "lib": ["ES2021"],
+        "target": "ES2021"
```
Comment (voicis) [inline packages/cds-odata-annotation-converter/tsconfig.json:8]: This one still has target set to `ES2021`. Was it intentional?

File: packages/cds-annotation-parser/tsconfig.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
         "outDir": "dist",
         "rootDir": "src",
         "lib": ["ES2021"],
+        "target": "ES2021",
```
Comment (voicis) [inline packages/cds-annotation-parser/tsconfig.json:7]: This one still has target set to `ES2021`. Was it intentional?

File: packages/cds-odata-annotation-converter/tsconfig.json (hunk 2 above)
Comment (devinea) [reply packages/cds-odata-annotation-converter/tsconfig.json:8]: Screenshot 2024-07-05 at 15 40 31
Tests fail when updated in annotation packages. 
All seem to show this line as https://github.com/SAP/open-ux-tools/blob/main/packages/cds-annotation-parser/src/parser/parser.ts#L345

Comment (voicis) [reply packages/cds-odata-annotation-converter/tsconfig.json:8]: ok, it looks like there is some issue with chevrotain with that target. We can look at it separately.

**Review by voicis — APPROVED** [review]
Control Property Editor changes looks good and tested it locally with node 18.

**Pull request merged**